src/application/services/auth.service.ts

```diff
@@ -25,17 +26,23 @@ class AuthService {
     return { ...userWithOutHash, token };
   }
 
-  async register(userProps: IUserProps) {
+  async register(registerDto: IRegisterDto) {
     const isUserWithSameEmailPresent =
-      await UserRepoInstance.userWithSameEmailExists(userProps.email);
+      await UserRepoInstance.userWithSameEmailExists(registerDto.email);
```

Comment:
use generic method like `findUser` or `fetchUser` that can accept object to perform the search on database.

Reply:
would do in future